src/_igraph/convert.c

```diff
@@ -2713,11 +2713,8 @@ int igraphmodule_PyObject_to_vector_list_t(PyObject* list, igraph_vector_list_t*
     return 1;
   }
 
-  /* TODO: what about this? is memory managed already? */
-  //IGRAPH_VECTOR_PTR_SET_ITEM_DESTRUCTOR(vec, igraph_vector_int_destroy);
   while ((item = PyIter_Next(it)) != 0) {
-    /* FIXME: is this even the right way to do this? */
-    subvec = igraph_Calloc(1, igraph_vector_t);
+    subvec = IGRAPH_CALLOC(1, igraph_vector_t);
```

Comment:
There is no longer any need to allocate this on the heap. Keep it on the stack.

It seems to me that there was a memory leak in the original function as the heap-allocated vector was not freed on error. https://github.com/igraph/python-igraph/issues/512 In this refactoring, this has been converted to a permanent memory leak (it happens regardless of error).

Reply:
I think I understand the memory leak, but not the heap/stack talk. What heap, what stack?

Reply:
instead of

```
igraph_vector_t *subvec;
subvec = IGRAPH_CALLOC(1, igraph_vector_t);
```

use

```
igraph_vector_t subvec;
```

There is no need for heap allocation.

Then there will also be no need for `free`ing what `IGRAPH_CALLOC` allocated.

Reply:
right, thank you